modules/porous_flow/src/dirackernels/PorousFlowLineGeometry.C

```diff
@@ -129,6 +135,64 @@ PorousFlowLineGeometry::PorousFlowLineGeometry(const InputParameters & parameter
     _half_seg_len[0] = _line_length;
 }
 
+void
+PorousFlowLineGeometry::regenPoints()
+{
+  if (!_point_file.empty())
+    return;
+
+  // recalculate the auto-generated points:
+  _rs.clear();
+  _xs.clear();
+  _ys.clear();
+  _zs.clear();
+
+  // add point for each cell the line passes through
+  Point p0(_line_base[1], _line_base[2], _line_base[3]);
+  Point p1 = p0 + _line_length * _line_direction / _line_direction.norm();
```

Comment:
Add an extra check to see if this point exists in the mesh up front so that you get a more meaningful out of bounds error?

Reply:
Not a bad idea.